.github/workflows/interop-test.yml

```diff
@@ -1,32 +1,37 @@
 name: Interoperability Testing
-
 on:
   pull_request:
   push:
     branches:
-      - master
-
-concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
+      - "master"
 
 jobs:
-  # NOTE: during a pull request run, github creates a merge commit referenced in `github.sha`
-  # that merge commit is not a regular commit. You won't find it with a regular `git checkout SHA` and
-  # tools like `go get repo@SHA` won't find it.
-  #
-  # As a workaround, we generate a path to the actual pull request's commit, it looks like:
-  # `github.com/external-org/go-libp2p@latest-commit-on-their-branch`
-  run-ping-interop-cross-version:
-    uses: "libp2p/test-plans/.github/workflows/run-composition.yml@master"
-    with:
-      composition_file: "ping/_compositions/rust-cross-versions.toml"
-      custom_git_target: github.com/${{ github.event.pull_request.head.repo.full_name || github.event.repository.full_name }}
-      custom_git_reference: ${{ github.event.pull_request.head.sha || github.sha }}
-  run-ping-interop-cross-implementation:
-    uses: "libp2p/test-plans/.github/workflows/run-composition.yml@master"
+  build-ping-container:
+    name: Build Ping interop container
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Build image
+        working-directory: ./test-plans
+        run: make
+      - name: Upload ping versions info
+        uses: actions/upload-artifact@v3
+        with:
+          name: ping-versions
+          path: ./test-plans/ping-versions.json
+      - name: Upload image tar
+        uses: actions/upload-artifact@v3
+        with:
+          name: ping-image
+          path: ./test-plans/ping-image.tar
+  run-multidim-interop:
+    name: Run multidimensional interoperability tests
+    needs: build-ping-container
+    uses: "libp2p/test-plans/.github/workflows/run-testplans.yml@master"
```

Comment:
I think we could save some complexity here if this were just a [composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) that we call after building the images rather than a separate job.

That would allow the action within `test-plans` to access local state like built docker images, configuration files without us having to up and download artifacts.

@MarcoPolo What was the reasoning for making this a workflow rather than a composite action?

Reply:
For example, within the same job, we wouldn't need to tar up the container because it would still be accessible by tag.

Reply:
Good point! I didn't realize I could make a composite action. My knowledge here was out of date. Thanks!

I'll try this out soon. I agree it'll be nice not to have to upload the image. And the upload artifacts are always directories, so it'll be nice to avoid that.